src/Storageless/Http/SessionMiddleware.php

```diff
@@ -47,7 +47,7 @@ final class SessionMiddleware implements MiddlewareInterface
     public const ISSUED_AT_CLAIM      = 'iat';
     public const SESSION_CLAIM        = 'session-data';
     public const SESSION_ATTRIBUTE    = 'session';
-    public const DEFAULT_COOKIE       = 'slsession';
+    public const DEFAULT_COOKIE       = '__Secure-slsession';
```

Comment:
BC Break!

Reply:
Hmm, given that the defaults (including this constant) already enforce `Secure` and `Same-Site`, what does this add?

Specifically, in order to have a non-secure cookie, somebody would have to design a blueprint cookie on their own and use the constructor manually.

Maybe a better solution would be to make the ctor `@internal`?

Reply:
Scenario:

1. User go to `http://example.com` (non-HTTPS)
1. `http://example.com` replies with 2 headers:
    1. `Location: https://example.com` (go to HTTPS)
    1. `Set-Cookie: slsession=XXX; Secure`
1. User's browser accepts the cookie
1. User's browser go to HTTPS version and sends the `slsession` cookie

A MITM can get the cookie during the HTTP-to-HTTPS redirect, and the browser doesn't reject it. `Secure` and  `Same-Site` flags are ineffective in this case.

Reply:
Oh, that sounds like a CVE in upstream though?

I think the patch overall makes sense, but is there a meaningful way to test this and to prevent regressions?

Reply:
> Oh, that sounds like a CVE in upstream though?

I'm not familiar with how CVEs should be handled, but I wouldn't consider this improvement a CVE of this library. Note that the RFC isn't a ratified standard yet.

> I think the patch overall makes sense, but is there a meaningful way to test this and to prevent regressions?

Define *meaningful* please :confounded: I've altered the test, should be enough, isn't it?

Reply:
I think what's missing is something to prevent a regression: the value of this default cookie name is relevant from a functional perspective, so even having a test that verifies that this prefix exists on the defaults (when using the two named ctors) is important.

Wording and documenting those tests (all there is to test is an `assertStringStartsWith...`) would include the results of this conversation, possibly in a docblock.

Potentially interesting to reference the above RFC draft in `README.md`?